dev-docs/design/how-lit-html-works.md

```diff
@@ -0,0 +1,426 @@
+
+An attribute on the element may also represent an `ElementPart` in the case where the attribute name starts with the marker sentinel value. E.g. `<div ${} ${}>` is marked up as `<div lit$random$0="" lit$random$1="">`.
+
+For each of the bound attributes identified on the element, they are removed from the element and a `TemplatePart` is created to record the location, name, and type of dynamic binding.
+
+##### Comment
+
+A comment that matches a marker annotation is a `TemplatePart` in ChildPart position.
+
+##### Text
+
+Usually text-position markers will be comment nodes, but inside `<style>`, `<script>`, `<textarea>`, `<title>` tags the markup that _looks_ like a comment (`<!-- lit -->`) will be inserted as _text_. So we must search for the marker within Text nodes. If found we split the Text node and insert a comment marker, and store a `TemplatePart` in ChildPart position referencing the comment marker's node index.
+
+### 2.ii. Create
+
+The create phase is performed only when rendering for **the first time to a specific container's ChildPart**.
+
+In the `lit-html` source code this phase is the [instantiating and cloning](https://github.com/lit/lit/blob/64fb960246e8f1eae982c2f09fca9759001756af/packages/lit-html/src/lit-html.ts#L1534-L1535) of a `TemplateInstance`.
+
+For the example, this phase results in the DOM nodes being cloned into a `fragment`, ready to be inserted in the document, and a list of `Parts` that can be updated by the next phase to set the dynamic JavaScript values in that fragment.
+
+```js
+const instance = new TemplateInstance(preparedTemplate);
+// The created DOM fragment will be inserted into the DOM.
+const fragment = instance._clone();
+
+console.log(instance._$parts);
+// [AttributePart, ChildPart, EventPart]
+```
+
+After the update phase this fragment is inserted into the DOM.
+
+The following sections cover in detail what happens when the `TemplateInstance` is created and then cloned:
+
+#### Create a `TemplateInstance` and instantiate `Part`s
+
+A `TemplateInstance` is responsible for creating the initial DOM and updating that DOM. It's an updatable instance of a `Template`. The `TemplateInstance` holds references to the `Part`s used to update the DOM.
+
+Within `TemplateInstance._clone()`, first the `<template>` element is cloned into a document fragment.
+
+After cloning, the document fragment node tree is walked to associate nodes with `TemplatePart`s by depth-first-index. When a node's index matches the index stored on a `TemplatePart`, a `Part` instance is created for the node. Based on the data in the `TemplatePart`, one of `ChildPart`, `AttributePart`, `PropertyPart`, `EventPart`, `BooleanAttributePart`, or `ElementPart` is instantiated.
```

Comment:
I think it's good to explicitly say the Part holds a reference to the Node so we get immediate access for updates.
```suggestion
After cloning, the document fragment node tree is walked to associate nodes with `TemplatePart`s by depth-first-index. When a node's index matches the index stored on a `TemplatePart`, a `Part` instance is created for the node which holds a reference to the node for direct access during updates. Based on the data in the `TemplatePart`, one of `ChildPart`, `AttributePart`, `PropertyPart`, `EventPart`, `BooleanAttributePart`, or `ElementPart` is instantiated.
```

Reply:
Nice!